app/src/main/contrib/actions/coverageActionCreators.ts

```diff
@@ -0,0 +1,59 @@
+import { Dispatch } from "redux";
+
+import {ContribAppState} from "../reducers/contribAppReducers";
+import {
+    CoverageDataSetsFetched,
+    CoverageOneTimeTokenClear, CoverageOneTimeTokenFetched, CoverageSetFormat,
+    CoverageTypes
+} from "../actionTypes/CoverageTypes";
+import {CoverageService} from "../services/CoverageService";
+
+export const coverageActionCreators = {
+
+    getDataSets() {
+        return async (dispatch: Dispatch<any>, getState: () => ContribAppState) => {
+
+            const touchstone = getState().touchstones.currentTouchstone;
+            const group = getState().groups.currentUserGroup;
+            const responsibility = getState().responsibilities.currentResponsibility;
+
+            const sets: any = await (new CoverageService(dispatch, getState))
+                .getDataSets(group.id, touchstone.id, responsibility.scenario.id);
+
+            return dispatch({
+                type: CoverageTypes.COVERAGE_DATA_SETS_FETCHED,
+                data: sets ? sets.coverage_sets: null
+            } as CoverageDataSetsFetched );
+        }
+    },
+
+    setFormat(format: string) {
+        return (dispatch: Dispatch<any>) => {
+            dispatch({
+                type: CoverageTypes.COVERAGE_SET_FORMAT,
+                data: format
+            } as CoverageSetFormat);
+        }
+    },
+
+    getOneTimeToken() {
+        return async (dispatch: Dispatch<any>, getState: any) => {
+            dispatch({
+                type: CoverageTypes.COVERAGE_ONE_TIME_TOKEN_CLEAR,
+            } as CoverageOneTimeTokenClear );
+
+            const touchstone = getState().touchstones.currentTouchstone;
+            const format = getState().coverage.selectedFormat;
+            const group = getState().groups.currentUserGroup;
+            const responsibility = getState().responsibilities.currentResponsibility;
```

Comment:
Looks like touchstone id, group id and scenario id are used both here and in the above function, so would be nice to refactor this out into a re-usable function like
```
function getResponsibilityIds(state: ContribAppState){
    const groupId = state.groups.currentUserGroup.id;
    const touchstoneId = state.touchstones.currentTouchstone.id;
    const scenarioId = state.responsibilities.currentResponsibility.scenario.id;
    return {groupId, touchstoneId, scenarioId}
}
```
This can be called like
```
const ids = getResponsibilityIds(getState());
 const token: string = await (new CoverageService(dispatch, getState))
+                .getOneTimeToken(ids.groupId, ids.touchstoneId, ids.scenarioId, format);

```

I think these 3 ids are also used in other action creators (e.g. I see them also in `runParamerersActionCreators`) so this function could live in some helper class accessible to all of them.